File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> map = new HashMap<>();
        for(String[] cloth : clothes){
            map.put(cloth[1], map.getOrDefault(cloth[1], 0)+1);
        }
        
        System.out.println(map);
                
        for(int num : map.values()){
            answer *= num + 1;
        }
        
        return answer-1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(string[,] clothes) {
        int answer = 1;
        var dict = new Dictionary<string, int>();
        
        for(int i=0; i<clothes.GetLength(0); i++){
            string category = clothes[i, 1];
            dict[category] = dict.GetValueOrDefault(category, 0) + 1;
        }
        
        foreach(var num in dict.Values){
            answer *= num + 1;
        }
        return answer-1;
    }
}